### PR TITLE
Mattermost implementation changes for botkube channel.

### PR DESCRIPTION
**Pull request by mugdha-adhav**
Removed CreateChannel support from Mattermost.
Formatted postMessage output.
Updated README for adding channel in Mattermost.

### CODE REVIEW

File: pkg/notify/mattermost.go [hunk 1]
```diff
@@ -31,18 +31,20 @@ func NewMattermost(c *config.Config) Notifier {
 	client.SetOAuthToken(c.Communications.Mattermost.Token)
 	botTeam, resp := client.GetTeamByName(c.Communications.Mattermost.Team, "")
 	if resp.Error != nil {
-		log.Logger.Fatal("Error in connecting to Mattermost team. Error: ", resp.Error)
+		log.Logger.Error("Error in connecting to Mattermost team. Error: ", resp.Error)
```
Comment (ssudake21) [inline pkg/notify/mattermost.go]: Also dump Team name in error.

File: pkg/notify/mattermost.go [hunk 2]
```diff
@@ -31,18 +31,20 @@ func NewMattermost(c *config.Config) Notifier {
...
+		return nil, resp.Error
 	}
 	botChannel, resp := client.GetChannelByName(c.Communications.Mattermost.Channel, botTeam.Id, "")
 	if resp.Error != nil {
-		log.Logger.Fatal("Error in connecting to Mattermost channel. Error: ", resp.Error)
+		log.Logger.Error("Error in connecting to Mattermost channel. Error: ", resp.Error)
```
Comment (ssudake21) [inline pkg/notify/mattermost.go]: Dump channel name in error

File: pkg/mattermost/mattermost.go [hunk 3]
```diff
@@ -176,10 +176,7 @@ func (b *Bot) getChannel() *model.Channel {
 	// Checking if channel exists
 	botChannel, resp := client.GetChannelByName(b.ChannelName, b.getTeam().Id, "")
 	if resp.Error != nil {
-		// Creating channel
-		if botChannel, resp = client.CreateChannel(mmChannel(b.ChannelName, b.getTeam().Id)); resp.Error != nil {
-			logging.Logger.Fatal("There was a problem finding channel in Mattermost. Error: ", resp.Error)
-		}
+		logging.Logger.Fatal("There was a problem finding channel in Mattermost. Error: ", resp.Error)
```
Comment (ssudake21) [inline pkg/mattermost/mattermost.go]: Dump all contextual data in error. Channel  name etc.

**Comment by PrasadG193** [comment]
Please rebase with develop

File: pkg/mattermost/mattermost.go (hunk 3 above)
Comment (mugdha-adhav) [reply pkg/mattermost/mattermost.go]: Made required changes.

File: pkg/notify/mattermost.go (hunk 2 above)
Comment (mugdha-adhav) [reply pkg/notify/mattermost.go]: Made required changes.

File: pkg/notify/mattermost.go (hunk 1 above)
Comment (mugdha-adhav) [reply pkg/notify/mattermost.go]: Made required changes.

File: pkg/mattermost/mattermost.go [hunk 4]
```diff
@@ -176,10 +176,7 @@ func (b *Bot) getChannel() *model.Channel {
 	// Checking if channel exists
 	botChannel, resp := client.GetChannelByName(b.ChannelName, b.getTeam().Id, "")
 	if resp.Error != nil {
-		// Creating channel
-		if botChannel, resp = client.CreateChannel(mmChannel(b.ChannelName, b.getTeam().Id)); resp.Error != nil {
-			logging.Logger.Fatal("There was a problem finding channel in Mattermost. Error: ", resp.Error)
-		}
+		logging.Logger.Fatal("There was a problem finding Mattermost channel ", b.ChannelName, "\nError: ", resp.Error)
```
Comment (PrasadG193) [inline pkg/mattermost/mattermost.go]: Would this also kill main routine?

File: config.yaml [hunk 5]
```diff
@@ -123,7 +123,7 @@ communications:
 
   # Settings for Mattermost
   mattermost:
-    enabled: true
+    enabled: false
     url: 'MATTERMOST_SERVER_URL'                # e.g https://example.com:9243
     token: 'MATTERMOST_TOKEN'
     team: 'MATTERMOST_TEAM'
```
Comment (PrasadG193) [inline config.yaml]: We should add one liner description for these fields

File: pkg/mattermost/mattermost.go (hunk 4 above)
Comment (mugdha-adhav) [reply pkg/mattermost/mattermost.go]: No it won't.

File: config.yaml (hunk 5 above)
Comment (mugdha-adhav) [reply config.yaml]: Made the required changes.

**Pull request merged**